lfs/transfer_queue.go

```diff
@@ -432,10 +491,8 @@ func (q *TransferQueue) errorCollector() {
 // retryCollector runs in its own goroutine.
 func (q *TransferQueue) retryCollector() {
 	for t := range q.retriesc {
-		q.rmu.Lock()
-		q.retryCount[t.Oid()]++
-		count := q.retryCount[t.Oid()]
-		q.rmu.Unlock()
+		q.rc.Increment(t.Oid())
+		count := q.rc.CountFor(t.Oid())
```

Comment:
The old behavior kept the same lock for both operations: add and retrieve. This isn't necessary, since only `retryCollector` is running at a time.